xtask/src/util.rs

```diff
@@ -257,53 +257,6 @@ pub fn read_dockerfiles(msg_info: &mut MessageInfo) -> cross::Result<Vec<(PathBu
     Ok(dockerfiles)
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    use cross::shell::Verbosity;
-    use std::collections::BTreeMap;
-
-    #[test]
-    fn check_ubuntu_base() -> cross::Result<()> {
-        // count all the entries of FROM for our images
-        let mut counts = BTreeMap::new();
-        let mut msg_info = Verbosity::Verbose.into();
-        let dockerfiles = read_dockerfiles(&mut msg_info)?;
-        for (path, dockerfile) in dockerfiles {
-            let lines: Vec<&str> = dockerfile.lines().collect();
-            let index = lines
-                .iter()
-                .map(|x| x.trim())
-                .position(|x| x.to_lowercase().starts_with("from"))
-                .ok_or_else(|| eyre::eyre!("unable to find FROM instruction for {:?}", path))?;
-            let tag = lines[index]
-                .split_whitespace()
-                .nth(1)
-                .ok_or_else(|| eyre::eyre!("invalid FROM instruction, got {}", lines[index]))?;
-            if let Some(value) = counts.get_mut(tag) {
-                *value += 1;
-            } else {
-                counts.insert(tag.to_string(), 1);
-            }
-        }
-
-        // Now, get the most common and ensure our base is correct.
-        let actual_base = cross::docker::UBUNTU_BASE;
-        let max_base = counts
-            .iter()
-            .max_by(|x, y| x.1.cmp(y.1))
-            .map(|(k, _)| k)
-            .ok_or_else(|| eyre::eyre!("have no dockerfiles"))?;
-
-        if actual_base != max_base {
-            eyre::bail!("most common base image is {max_base} but source code has {actual_base}")
-        } else {
-            Ok(())
-        }
-    }
-}
```

Comment:
Shouldn't this check the base image instead?

Reply:
Fixed as of the latest builds, which checks the base in `Dockerfile.ubuntu` is the same as the `CROSS_IMAGE` version.